src/libraries/System.Formats.Tar/tests/TarEntry/GnuTarEntry.Tests.cs

```diff
@@ -93,5 +94,343 @@ public void SupportedEntryType_Fifo()
             SetFifo(fifo);
             VerifyFifo(fifo);
         }
+
```

Comment:
Does it make sense to have tests that validate the offset is correct for an archive with multiple entries (reading)?

Reply:
Well, the current tests already confirm this is working as expected: When reading a PAX entry, the actual entry will always be preceded by an extended attributes entry, which we _don't_ return to the user, but rather read internally, collect all the extended attribute data, and then make it available in the actual entry (the second one) via a public dictionary property. So this means we are successfully reading two entries and returning the data offset of the second entry.

Similarly, in GNU long link and long path entries, the actual entry is preceded by either one or two metadata entries that are not returned to the user. I made sure to add a test for long link (when the link target path is too long to fit in the standard attribute field), long path (when a regular file's path is too long to fit in the standard attribute field) and a long link + long path test (when it's a link and both the link path and the link target path are too long to fit in the standard metadata fields). This last test is the one that actually reads 2 metadata entries before reading the 3rd one which is the actual file, and I confirmed the data offset position belongs to that third entry.

Reply:
I'm just looking for a test that asserts the DataOffset value for more than one entry since I suspect the use cases will be for real world archives that have many entries.  Maybe we could do that for one of the tar's that's part of test data?

Reply:
Added tests that write two entries and verify the offset value of the first and second entry